### PR TITLE
autoconf Dependency Fix

### PR DESCRIPTION
**Pull request by jcsiadal**
Fixes missing install dependencies.
Also made some formatting changes to the specfile.

### CODE REVIEW

File: components/dev-tools/autoconf/SPECS/autoconf.spec [hunk 1]
```diff
@@ -24,11 +24,18 @@ Source0:   https://ftp.gnu.org/gnu/autoconf/autoconf-%{version}.tar.gz
...
+BuildRequires: perl
 BuildRequires: perl-macros
 BuildRequires: perl(File::Compare)
 BuildRequires: perl(File::Copy)
 BuildRequires: perl(Data::Dumper)
 BuildRequires: perl(Text::ParseWords)
+%if 0%{?suse_version} || 0%{?sle_version}
+Requires: perl-base
+%else
+Requires: perl-interpreter
```
Comment (adrianreber) [inline components/dev-tools/autoconf/SPECS/autoconf.spec:36]: The perl-base or perl-interpreter explicit dependency seems unnecessary. `perl(File::Compare)` pulls in those packages on each platform already.

File: components/dev-tools/autoconf/SPECS/autoconf.spec [hunk 2]
```diff
@@ -48,37 +55,42 @@ may be configuring software with an Autoconf-generated script;
...
+
 
 %install
 make %{?_smp_mflags} DESTDIR=$RPM_BUILD_ROOT install
 
 # remove share/info/dir to avoid conflict with other package installs
 rm -f $RPM_BUILD_ROOT/%{install_path}/share/info/dir
 
-%{__mkdir_p} ${RPM_BUILD_ROOT}/%{_docdir}
+mkdir -p ${RPM_BUILD_ROOT}/%{_docdir}
```
Comment (adrianreber) [inline components/dev-tools/autoconf/SPECS/autoconf.spec:79]: Why this change?

**Comment by adrianreber** [comment]
> Also made some formatting changes to the specfile.

If you want to make formatting changes please add them to separate commits and do not mix functional changes with formatting changes.

File: components/dev-tools/autoconf/SPECS/autoconf.spec (hunk 1 above)
Comment (jcsiadal) [reply components/dev-tools/autoconf/SPECS/autoconf.spec:36]: I looked through the dependency chain and it doesn't appear that RHEL's perl-file-compare will install the perl binary. I also looked at the Fedora autoconf specfile and they explicitly add perl-interpreter. The SUSE one isn't needed, but I wanted to keep things parallel.

File: components/dev-tools/autoconf/SPECS/autoconf.spec (hunk 2 above)
Comment (jcsiadal) [reply components/dev-tools/autoconf/SPECS/autoconf.spec:79]: I'm surprised you would ask. Red Hat's RPM guide (at least one of them) says not to use (LSB) shell command macros. They were put in to make RPM portable across OSs (like Windows or BSD) and that never happened. They're just extra work with no benefit. SUSE says to never use them.
When I touch an RPM spec, I try and clean it up to current RPM standards, i.e. both RH and SUSE mandate %license now.

File: components/dev-tools/autoconf/SPECS/autoconf.spec (hunk 1 above)
Comment (adrianreber) [reply components/dev-tools/autoconf/SPECS/autoconf.spec:36]: If I do `dnf -y install "perl(File::Compare)"` `perl-interpreter` is pulled in on 9.1.

Comment (jcsiadal) [reply components/dev-tools/autoconf/SPECS/autoconf.spec:36]: > "you SHOULD NOT assume any other packages are present as RPM dependencies and anything brought into the buildroot by the build system can change over time." 

That's from Red Hat's packaging guidelines. They do say that one can usually drop Requires: if we're running autorequires, but I thought we're only doing manual deps. The other question to ask: if it's not needed, why does Fedora include it in their autoconf spec?
